src/execution/store.rs

```diff
@@ -557,65 +562,170 @@ impl<'b, T> Store<'b, T> {
             panic!("Invalid parameters for function");
         }
 
-        match &func_inst {
-            FuncInst::HostFunc(host_func_inst) => {
-                let returns = (host_func_inst.hostcode)(&mut self.user_data, params);
-                debug!("Successfully invoked function");
-
-                // Verify that the return parameters match the host function parameters
-                // since we have no validation guarantees for host functions
-
-                let return_types = returns.iter().map(|v| v.to_ty()).collect::<Vec<_>>();
-                if func_ty.returns.valtypes != return_types {
-                    trace!(
-                        "Func param types len: {}; Given args len: {}",
-                        func_ty.params.valtypes.len(),
-                        param_types.len()
-                    );
-                    return Err(RuntimeError::HostFunctionSignatureMismatch);
-                }
+        Ok(ResumableRef::Fresh(FreshResumableRef {
+            func_addr,
+            params,
+            maybe_fuel,
+        }))
+    }
+
+    pub fn resume(&mut self, resumable_ref: ResumableRef) -> Result<RunState, RuntimeError> {
+        match resumable_ref {
+            ResumableRef::Fresh(FreshResumableRef {
+                func_addr,
+                params,
+                maybe_fuel,
+            }) => {
+                // Err can never happen here
+                let func_inst = self
+                    .functions
+                    .get(func_addr)
+                    .ok_or(RuntimeError::FunctionNotFound)?;
```

Comment:
```suggestion
                let func_inst = self
                    .functions
                    .get(func_addr)
                    .expect("func addrs to always be valid if the correct store is used")
```

Let's panic for now. The planned `Stored<T>` object will fix these issues.